python/ray/train/callbacks/callback.py

```diff
@@ -5,16 +5,8 @@
 class TrainingCallback(abc.ABC):
     """Abstract Train callback class."""
 
-    def handle_result(self, results: List[Dict], **info):
-        """Called every time train.report() is called.
-
-        Args:
-            results (List[Dict]): List of results from the training
-                function. Each value in the list corresponds to the output of
-                the training function from each worker.
-            **info: kwargs dict for forward compatibility.
-        """
-        pass
+    def __init__(self):
```

Comment:
Should TrainingCallbacks also accept a list of preprocessors as an arg to `__init__`?

For example, if a user wants to print the results of worker 0, they can then do so easily via `PrintCallback(preprocessors=[WorkersResultsPreprocessor(workers_to_log=0)])`

instead of having to do the following

```
class WorkerZeroPrintCallback(PrintCallback):
    def __init__(self):
        self._result_preprocessors = [WorkersResultsPreprocessor(workers_to_log=0)]
```

Reply:
Yeah! Let me think about the implementation more though as this ultimately depends on the overall structure of how we handle preprocessing.

In general we should definitely allow the user to configure their own preprocessors, but I'm not sure if this should be done through `__init__` or some setter.

One thing to think about is that it probably doesn't make sense to allow this to be set for Callbacks that don't implement `handle_results`.

Reply:
That should still be fine even if `handle_results` is not implemented right? Ultimately it would just be a no-op then.

Reply:
Yeah I don't disagree, my comment was more so to consider if there is a cleaner way of tying everything together (e.g. if we end up using a mixin for preprocessing then it would make sense to tie all the preprocessor logic to the mixin itself rather the base class).